src/nvim/ex_cmds.c

```diff
@@ -4960,3 +4961,27 @@ void ex_oldfiles(exarg_T *eap)
     }
   }
 }
+
+void ex_trust(exarg_T *eap)
+{
+  const char *const p = skiptowhite(eap->arg);
+  char *arg1 = xmemdupz(eap->arg, (size_t)(p - eap->arg));
+  const char *action = "allow";
+  const char *path = eap->arg;
+
+  if (strcmp(arg1, "++deny") == 0) {
+    path = skipwhite(p);
+    action = "deny";
+  } else if (strcmp(arg1, "++remove") == 0) {
+    path = skipwhite(p);
+    action = "remove";
+  }
```

Comment:
Using `:trust {file}` gives a confusing error message since we are not checking for that here:

```
Error executing vim.secure.trust: runtime/lua/vim/secure.lua:174: bufnr is required when action is "allow"
stack traceback:
        [C]: in function 'assert'
        runtime/lua/vim/secure.lua:174: in function <runtime/lua/vim/secure.lua:135>
```

I think we should add a check here to catch cases where someone uses `:trust {file}`, because it is not at all obvious what that error means or how to fix it.

```suggestion
  if (strcmp(arg1, "++deny") == 0) {
    path = skipwhite(p);
    action = "deny";
  } else if (strcmp(arg1, "++remove") == 0) {
    path = skipwhite(p);
    action = "remove";
  } else if (*arg1 != '\0') {
    semsg(e_invarg2, arg1);
    xfree(arg1);
    return;
  }
```

This will instead show `E475: Invalid argument: {file}`, which is slightly clearer.

Reply:
We could now initialize `path` to `skipwhite(p)` on `:4970` with this change. Also are such early returns okay? I mostly saw `goto` being used.

Reply:
Sure go for it.

>Also are such early returns okay? I mostly saw goto being used.

goto would be useful to avoid duplicating `xfree(arg1)`, so go ahead and add that too.

Reply:
done in 5653a043dca1508ff5797f67231c1a90f771e25c